### PR TITLE
workflow yml improvements

### PR DESCRIPTION
**Pull request by Margen67**
```
ci.yml:
 Use job-wide shell: bash.
 Always use checkout v2.
 Upgrade commitlint-github-action to v3.
 Remove brackets around check-code-formatting since it's only one item.
 Replace MinSizeRel with Release.
 upload-artifact:
  Always use v2.
  name:
   Remove unneeded quotation marks around name.
   Use runner.os when possible.
  if-no-files-found: error
 Make Windows mingw win32/NT5.1 into matrix.
 Add HOMEBREW_NO_ANALYTICS=1 to brew install.
 Make linux-portable(-32) into matrix.
 linux-docker:
  Use checkout action instead of git clone.
localisation.yml:
 Remove erroneous space after cron:
 Use --depth 1 to speed up cloning Localisation.
 Commit and push:
  Replace pushd with working-directory.
  Remove unneeded popd.
```

### CODE REVIEW

**Review by tupaschoal — COMMENTED** [review]
Changes like this are very welcome, thank you!

If you can make smaller, more self-contained commits though, it would be great. Reviewing it all at once, even though you helpfully describe all changes in the commit message, sometimes becomes confusing. Note it still could all be in one PR, but with commits like:

- Upgrade checkout action to v2
- Use default bash by default
- ...

**Comment by Margen67** [comment]
> 
> 
> Changes like this are very welcome, thank you!
> 
> If you can make smaller, more self-contained commits though, it would be great. Reviewing it all at once, even though you helpfully describe all changes in the commit message, sometimes becomes confusing. Note it still could all be in one PR, but with commits like:
> 
>     * Upgrade checkout action to v2
> 
>     * Use default bash by default
> 
>     * ...

@tupaschoal Done.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -77,47 +77,42 @@ jobs:
...
         uses: hendrikmuhs/ccache-action@v1
         with:
-          key: windows-mingw-nt51
+          key: ${{ matrix.cache_key }}
       - name: Build OpenRCT2
         run: |
           sudo su
           mkdir bin && cd bin
-          cmake .. -G Ninja -DCMAKE_TOOLCHAIN_FILE=../CMakeLists_mingw.txt -DCMAKE_BUILD_TYPE=MinSizeRel -DDISABLE_IPO=on -DDISABLE_HTTP=Off -DFORCE32=on -DENABLE_SCRIPTING=ON -DCMAKE_CXX_FLAGS="-Wno-error=cast-function-type -Wno-error=unused-function" -DSTATIC=on -DMINGW_TARGET_NT5_1=ON
+          cmake .. -G Ninja -DCMAKE_TOOLCHAIN_FILE=../CMakeLists_mingw.txt -DCMAKE_BUILD_TYPE=Release -DDISABLE_IPO=on -DFORCE32=on ${{ matrix.build_flags }}
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:109]: Why the change from `MinSizeRel` to `Release`? (this is on commit https://github.com/OpenRCT2/OpenRCT2/pull/14479/commits/b513c0f297bdd394d65cdc728bda7a29128f2ca4)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -158,7 +158,7 @@ jobs:
           key: macos
       - name: Build OpenRCT2
         run: |
-          brew install ninja
+          HOMEBREW_NO_ANALYTICS=1 brew install ninja
```
Comment (tupaschoal) [inline .github/workflows/ci.yml:161]: Why? What does it bring to the table? Maybe add the reasoning to the commit body

**Review by tupaschoal — COMMENTED** [review]
I have just a few comments. It seems fine, the commit introducing the matrix is the one that concerns me most, though they look fine.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (Margen67) [reply .github/workflows/ci.yml:109]: I thought `MinSizeRel` optimized for size instead of speed.

Comment (tupaschoal) [reply .github/workflows/ci.yml:109]: From what I searched it seems to try to do both, but people do recommend `Release` indeed, so I'm ok with this.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tupaschoal) [reply .github/workflows/ci.yml:161]: What about this one?

**Review by tupaschoal — APPROVED** [review]
This looks fine to me, anyone else on @OpenRCT2/developers  wanna take a look?

**Review by janisozaur — APPROVED** [review]
Looks about right, the only thing in weary of is renaming the upload artifacts, that needs to be monitored post-merge
